app/javascript/leaflet/maps/search_results_map.js

```diff
@@ -0,0 +1,150 @@
+import { latLngBounds } from "leaflet";
+import GeoBlacklightViewerMap from "../viewers/map";
+import GeoSearchControl from "../controls/geosearch";
+import { bboxToBounds, geoJSONToBounds } from "../utils";
+
+export default class SearchResultsMap {
+  constructor(element) {
+    const data = element.dataset,
+      opts = { baseUrl: data.catalogPath },
+      world = latLngBounds([
+        [-90, -180],
+        [90, 180],
+      ]);
+    let bbox;
+
+    // StaticButton
+    // Create the anchor element
+    const staticButtonNode = document.createElement("a");
+    staticButtonNode.setAttribute("id", "gbl-static-button");
+    staticButtonNode.setAttribute("href", "#");
+    staticButtonNode.setAttribute("style", "display:none;");
+    staticButtonNode.className = "btn btn-primary";
+    staticButtonNode.textContent = "Redo search here";
+
+    // Create the span element for the glyphicon
+    const span = document.createElement("span");
+    span.className = "glyphicon glyphicon-repeat";
+
+    // Append the span to the anchor element
+    staticButtonNode.appendChild(span);
+
+    // DynamicButton
+    const dynamicButtonNode = document.createElement("label");
+    dynamicButtonNode.setAttribute("id", "gbl-dynamic-button");
+    dynamicButtonNode.textContent = " Search when I move the map";
+    dynamicButtonNode.setAttribute("style", "display:block;");
+
+    // Create the input (checkbox) element
+    const input = document.createElement("input");
+    input.setAttribute("type", "checkbox");
+    input.checked = true; // Set the checkbox to be checked by default
+
+    // Insert the input element at the beginning of the label element
+    dynamicButtonNode.insertBefore(input, dynamicButtonNode.firstChild);
+
+    const initialOptions = {
+      dynamic: true,
+      baseUrl: "",
+      searcher: function () {
+        window.location.href = this.getSearchUrl();
+      },
+      delay: 800,
+      staticButton: staticButtonNode,
+      dynamicButton: dynamicButtonNode,
+    };
+
+    if (typeof data.mapGeom === "string") {
+      bbox = geoJSONToBounds(JSON.parse(data.mapGeom));
+    } else {
+      document
+        .querySelectorAll(".document [data-geom]")
+        .forEach((docElement) => {
+          try {
+            const geomData = JSON.parse(docElement.dataset.geom);
+            const currentBounds = geoJSONToBounds(geomData);
+            if (!world.contains(currentBounds)) {
+              throw new Error("Invalid bounds");
+            }
+            bbox =
+              typeof bbox === "undefined"
+                ? currentBounds
+                : bbox.extend(currentBounds);
+          } catch (e) {
+            bbox = bboxToBounds("-180 -90 180 90");
+          }
+        });
+    }
+
```

Comment:
This is missing a history statechange https://github.com/geoblacklight/geoblacklight/blob/24aba2e82f4b386e0301d8beebf969727bcf7fb0/app/assets/javascripts/geoblacklight/modules/results.js#L2-L9 https://github.com/geoblacklight/geoblacklight/blob/24aba2e82f4b386e0301d8beebf969727bcf7fb0/app/assets/javascripts/geoblacklight/modules/results.js#L37-L45 not 100% sure what it is doing and maybe it got removed because all browsers support his now?